kuma/payments/tests/test_utils.py

```diff
@@ -29,7 +29,7 @@ def test_get_stripe_customer_data(mock_retrieve):
 @mock.patch("stripe.Subscription.retrieve", return_value=mock.Mock(spec_set=["delete"]))
 def test_cancel_stripe_customer_subscription(mock_sub, mock_cust):
     """A stripe customer's subscriptions can be cancelled."""
-    cancel_stripe_customer_subscription("cust_id")
+    list(cancel_stripe_customer_subscription("cust_id"))
```

Comment:
This is rather ugly but there's a good reason for it. Lemme explain. 

First of, look at `payments/views.py` around line 51. If you don't do that pattern you don't get information about which subscriptions get canceled. It just cancels everything based on a customer :(
So that's the primary use case. for why `cancel_stripe_customer_subscription` is now a generator. And because it's a generator, nothing happens unless you "exhaust" it. 

What I'd *ideally* like is two-fold, and I don't know what should come first. 
On the "MDN subscription management" page the cancel button is for ALL your subscription. Another way to phrase it is that it cancels the whole customer. It would be better if that UI was one cancel button per subscription. It's silly that someone would have more than 1 subscriptions. If they have a $5.00 and a $7.50 something's screwy because it ought to be just one for $12.50. 
Either way, the Django view or the template UX should be done on a per-subscription basis. So what I *could* do is rewrite `payments/views.py` so that canceling a subscription triggers a Django signal. 

However, and here's the point! I didn't want to change `payments/views.py` too much. Not right now at least. Hence this one little change only. Just to get my code unblocked.

Reply:
What speaks against having `cancel_stripe_customer_subscription` just return a list of IDs for the cancelled subscriptions? Generators seem a little overblown to me here.

Reply:
Because I couldn't trust the `stripe` Python API. What if this happens:

```
>>> s = stripe.Subscription.retrieve('myid')
>>> s.id
'myid'
>>> s.delete()
>>> s.id
None
```
Maybe. Like in the Django ORM, if you delete an object, I think its primary key becomes `None` afterwards before Python "garbage collects". 

Also, I think I had in mind the idea that perhaps we want to trigger a signal BEFORE we delete it and one AFTER. 

BUT! Now I realize something stupid. The generator doesn't return `s.id` after the `s = stripe.Subscription.retrieve(()` call. It returns the `sub['id']` which most likely won't mutate. 

In conclusion, I think I definitely overcomplicated things. In the heat of the moment, I think I had grand ideas about refactoring some of the `payments` stuff but you're right, this can just return a list. I'll do that.

Reply:
Well, actually my intentions weren't that bad. You give it the customer ID and get a list of subscription *objects*. 
Either way, as of right now, none of the users of `cancel_stripe_customer_subscription` actually need the subscription *objects*. They're fine with it just being the IDs.